From c9d58f467cf1fc1abbac80e1afae8159bdcca10f Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Wed, 12 Sep 2018 15:53:35 -0700
Subject: [PATCH] TRITON-755 Want ability to filter CMON discovery results by
 tag (fix pegjs usage, ensure cannot publish without rendering pegjs files)

---
 CHANGES.md   |  5 +++++
 Makefile     | 33 ++++++++++++++++++---------------
 package.json |  5 +++--
 3 files changed, 26 insertions(+), 17 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index c3dfbf7..065788a 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -4,6 +4,11 @@
 
 (nothing yet)
 
+## 1.2.2
+
+- TRITON-755 Fix the release process (`make cutarelease`) to ensure pegjs files
+  are built and included in the published files.
+
 ## 1.2.1
 
 - TRITON-755: Fix missing peg.js generated files
diff --git a/Makefile b/Makefile
index b0bd378..843e4ab 100644
--- a/Makefile
+++ b/Makefile
@@ -17,21 +17,29 @@ JSL_FILES_NODE	 = $(JS_FILES)
 JSSTYLE_FILES	 = $(JS_FILES)
 JSSTYLE_FLAGS	 = -f tools/jsstyle.conf
 CLEAN_FILES += ./node_modules
+PEGJS	= node_modules/.bin/pegjs
 
 include ./tools/mk/Makefile.defs
 
+
 #
 # Targets
 #
+
 .PHONY: all
-all: $(PEGJS_FILES)
+all $(PEGJS):
 	npm install
 
-PEGJS		= node_modules/.bin/pegjs
-$(PEGJS):
-	npm install pegjs
+# The phony "pegjsfiles" target is here for the "prepublish" npm script to call
+# to ensure pegjs files are rendered before release.
+.PHONY: pegjsfiles
+pegjsfiles: $(PEGJS_FILES)
+
 %.js: %.pegjs $(PEGJS)
-	$(PEGJS) $< $@
+	$(PEGJS) $< > $@
+
+CLEAN_FILES += $(PEGJS_FILES) triton-tags-*.tgz
+
 
 .PHONY: test
 test: all
@@ -41,23 +49,18 @@ test: all
 test-in-parallel: all
 	NODE_NDEBUG= prove -j15 -e ./node_modules/.bin/tape test/*.test.js
 
-.PHONY: clean
-clean::
-	rm -f $(PEGJS_FILES)
-	rm -f triton-tags-*.tgz
-
-check:: versioncheck
 
 # Ensure CHANGES.md and package.json have the same version.
-.PHONY: versioncheck
-versioncheck:
+.PHONY: check-version
+check-version:
 	@echo version is: $(shell cat package.json | json version)
 	[[ `cat package.json | json version` == `grep '^## ' CHANGES.md | head -2 | tail -1 | awk '{print $$2}'` ]]
 
-check:: versioncheck
+check:: check-version
 
+# Publish a release to npm properly.
 .PHONY: cutarelease
-cutarelease: $(COMPLETION_FILE) versioncheck
+cutarelease: pegjsfiles check-version
 	[[ -z `git status --short` ]]  # If this fails, the working dir is dirty.
 	@which json 2>/dev/null 1>/dev/null && \
 	    ver=$(shell json -f package.json version) && \
diff --git a/package.json b/package.json
index be9d1e0..54e0104 100644
--- a/package.json
+++ b/package.json
@@ -1,18 +1,19 @@
 {
   "name": "triton-tags",
   "description": "Triton tags parsing and validation",
-  "version": "1.2.1",
+  "version": "1.2.2",
   "author": "Joyent (joyent.com)",
   "dependencies": {
     "assert-plus": "^1.0.0"
   },
   "devDependencies": {
-    "pegjs": "^0.9.0",
+    "pegjs": "0.10.0",
     "tape": "^4.5.1",
     "vasync": "^1.6.3"
   },
   "main": "./lib",
   "scripts": {
+    "prepublish": "make pegjsfiles",
     "test": "make test"
   },
   "repository": {
-- 
2.21.0

